nipype/pipeline/plugins/slurmgraph.py

```diff
@@ -0,0 +1,158 @@
+    )
+    hash_exists = False
+    try:
+        hash_exists, _, _, _ = checknode.hash_exists()
+    except Exception:
+        hash_exists = False
+    return (hash_exists and node_state_does_not_require_overwrite )
+
+
+class SLURMGraphPlugin(GraphPluginBase):
+    """Execute using SLURM
+
+    The plugin_args input to run can be used to control the SGE execution.
+    Currently supported options are:
+
+    - template : template to use for batch job submission
+    - qsub_args : arguments to be prepended to the job execution script in the
+                  qsub call
+
+    """
+    _template="#!/bin/bash"
+
+    def __init__(self, **kwargs):
+        if 'plugin_args' in kwargs and kwargs['plugin_args']:
+            if 'retry_timeout' in kwargs['plugin_args']:
+                self._retry_timeout = kwargs['plugin_args']['retry_timeout']
+            if  'max_tries' in kwargs['plugin_args']:
+                self._max_tries = kwargs['plugin_args']['max_tries']
+            if 'template' in kwargs['plugin_args']:
+                self._template = kwargs['plugin_args']['template']
+                if os.path.isfile(self._template):
+                    self._template = open(self._template).read()
+            if 'sbatch_args' in kwargs['plugin_args']:
+                self._sbatch_args = kwargs['plugin_args']['sbatch_args']
+            if 'dont_resubmit_completed_jobs' in kwargs['plugin_args']:
+                self._dont_resubmit_completed_jobs = plugin_args['dont_resubmit_completed_jobs']
+            else:
+                self._dont_resubmit_completed_jobs = False
+        super(SLURMGraphPlugin, self).__init__(**kwargs)
+
+    def _submit_graph(self, pyfiles, dependencies, nodes):
+        def make_job_name(jobnumber, nodeslist):
+            """
+            - jobnumber: The index number of the job to create
+            - nodeslist: The name of the node being processed
+            - return: A string representing this job to be displayed by SLURM
+            """
+            job_name='j{0}_{1}'.format(jobnumber, nodeslist[jobnumber]._id)
+            # Condition job_name to be a valid bash identifier (i.e. - is invalid)
+            job_name=job_name.replace('-','_').replace('.','_').replace(':','_')
+            return job_name
+        batch_dir, _ = os.path.split(pyfiles[0])
+        submitjobsfile = os.path.join(batch_dir, 'submit_jobs.sh')
+
+        cache_doneness_per_node = dict()
+        if self._dont_resubmit_completed_jobs: ## A future parameter for controlling this behavior could be added here
+            for idx, pyscript in enumerate(pyfiles):
+                node = nodes[idx]
+                node_status_done = node_completed_status(node)
+                ## If a node has no dependencies, and it is requested to run_without_submitting
+                ## then run this node in place
+                if (not node_status_done) and (len(dependencies[idx]) == 0 ) and (node.run_without_submitting == True):
```

Comment:
run_without_submitting shouldn't apply to a graph plugin - that option is to ensure nodes can run on the monitoring/submitting node. in the graph plugin case, since the scheduler takes over, this check would be unnecessary. i know this was taken from the SGEgraph code (and i didn't review it closely enough :( ).